E0558 needs to be updated to new format

From: src/test/compile-fail/E0558.rs

E0558 needs a help converted to span_label, updating this:

```
error[E0558]: export_name attribute has invalid format
  --> src/test/compile-fail/E0558.rs:11:1
   |
11 | #[export_name] //~ ERROR E0558
   | ^^^^^^^^^^^^^^
   |
   = help: use #[export_name="*"]
```

To:

```
error[E0558]: export_name attribute has invalid format
  --> src/test/compile-fail/E0558.rs:11:1
   |
11 | #[export_name] //~ ERROR E0558
   | ^^^^^^^^^^^^^^ did you mean #[export_name="*"]?
```

Hi @jonathandturner ,

I am stuck at this error while compiling. Tried cargo update on liblog, libserialize and libstd. But still facing the issue

```
Compiling serialize v0.0.0 (file:///Users/abhishek/code/rust/src/libserialize)
error[E0460]: found possibly newer version of crate `std` which `log` depends on
  --> src/libserialize/lib.rs:41:14
   |
41 | #[macro_use] extern crate log;
   |              ^^^^^^^^^^^^^^^^^
   |
   = note: perhaps that crate needs to be recompiled?
   = note: crate `std` path #1: /Users/abhishek/code/rust/build/x86_64-apple-darwin/stage0-sysroot/lib/rustlib/x86_64-apple-darwin/lib/libstd-9a82330aa9f25cb1.rlib
   = note: crate `std` path #2: /Users/abhishek/code/rust/build/x86_64-apple-darwin/stage0-sysroot/lib/rustlib/x86_64-apple-darwin/lib/libstd-9a82330aa9f25cb1.dylib
   = note: crate `log` path #1: /Users/abhishek/code/rust/build/x86_64-apple-darwin/stage0-rustc/x86_64-apple-darwin/release/deps/liblog-23e007e69714d505.dylib
   = note: crate `log` path #2: /Users/abhishek/code/rust/build/x86_64-apple-darwin/stage0-rustc/x86_64-apple-darwin/release/deps/liblog-23e007e69714d505.rlib

error: aborting due to previous error

error: Could not compile `serialize`.
```

--- a/src/libsyntax/attr.rs
+++ b/src/libsyntax/attr.rs
@@ -438,8 +438,9 @@ pub fn find_export_name_attr(diag: &Handler, attrs: &[Attribute]) -> Option<Inte
             } else {
                 struct_span_err!(diag, attr.span, E0558,
                                  "export_name attribute has invalid format")
-                                .help("use #[export_name=\"*\"]")
-                                .emit();
+                    .span_label(attr.span,
+                                &format!("did you mean #[export_name=\"*\"]?"))
+                    .emit();
                 None
             }
         } else {

--- a/src/test/compile-fail/E0558.rs
+++ b/src/test/compile-fail/E0558.rs
@@ -8,7 +8,10 @@
 // option. This file may not be copied, modified, or distributed
 // except according to those terms.
 
-#[export_name] //~ ERROR E0558
+#[export_name]
+//~^ ERROR E0558
+//~| NOTE did you mean #[export_name="*"]?
+
 pub fn something() {}
 
 fn main() {}

